### DISCUSSION TITLE
Dockerfile seems to be corrupted/missing in a new Docker image

### DISCUSSION THREAD

**Discussion by d31337ed** [umputun/remark42#1721] [Q&A]
Hi there!

I've tried to slightly change settings in my copy of `docker-compose.yml` and restart existing instances of remark42 container like I did few times before 

However, today I faced an error:

```
Building remark
[+] Building 0.0s (2/2) FINISHED                                                                                                                   docker:default
 => [internal] load build definition from Dockerfile                                                                                                         0.0s
 => => transferring dockerfile: 2B                                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                            0.0s
 => => transferring context: 2B                                                                                                                              0.0s
ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount1824847275/Dockerfile: no such file or directory
ERROR: Service 'remark' failed to build : Build failed
```

To eliminate an error I've tried to:
-  clear cache (`docker system prune -a`); 
- delete and download image manually from Dockerhub (`docker pull umputun/remark42`);
- build image from clean `docker-compose.yml` 
But the issue still appears.

For double check I've tried to execute a clean installation from docker image in two independent systems. Having the same issue on a second system:

```
[+] Building 0.0s (2/2) FINISHED                                                                                                              docker:desktop-linux
 => [remark internal] load .dockerignore                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                               0.0s
 => [remark internal] load build definition from Dockerfile                                                                                                   0.0s
 => => transferring dockerfile: 2B                                                                                                                            0.0s
failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount1797421555/Dockerfile: no such file or directory
```

**Steps to reproduce:**
1. Download clean `docker-compose.yml` from repo: https://github.com/umputun/remark42/blob/master/docker-compose.yml
2. Execute `docker-compose pull && docker-compose up -d`

Please review a content of the latest docker image
Thank you!

**Comment by umputun** [comment]
I'm uncertain about the meaning of "Dockerfile seems to be corrupted/missing in a new Docker image." The idea of a Dockerfile missing from Docker images is somewhat confusing, as the image is an artifact built using the Dockerfile, which doesn't include the Dockerfile itself.

Here's how you can download Docker Compose and start the container:

<img width="1566" alt="image" src="https://github.com/umputun/remark42/assets/535880/53a132ec-28fd-412c-9a00-5684f076663a">

As demonstrated, it works seamlessly.

Your first example seems to show an attempt to build from the source, rather than pulling the image. To build it, you should first clone the repository. The section where you mention "To eliminate an error, I've tried to..." also suggests an attempt to build the image, which isn't necessary if you pulled the image first and the image in your compose matches the published one.

Could there be an issue accessing the Docker Hub hosting the image? If so, you might consider changing the image in your compose to one stored on the GitHub registry, such as `ghcr.io/umputun/remark42:latest`.

**Comment by umputun** [comment]
as this is clearly not a bug but rather some sort of confusion, I'll move it to discussions

**Comment by d31337ed** [comment]
Oh, sorry for misplacing, it is definitely not a bug, you are right

After further investigation I've found out something is wrong with a new version of Docker BuildKit
Running container from image using `docker run` with manual passing of arguments works great 
Unfortunately I haven't found out what's wrong with BuildKit, but I hope this small workaround will be useful for someone who may also face that rouble 

Thank you so much and happy new year!
